.gitattributes

```
@@ -0,0 +1 @@
+* text=auto eol=lf
```

Let's make it specific instead?
```suggestion
/src/init/starship.* text eol=lf
```

---

Thanks for the suggestion. I also added `*.sh` for any other shell scripts and the `mod.rs` file (the bash init has multiple lines).

---

> and the `mod.rs` file (the bash init has multiple lines).

Isn't this ensured by rustc itself? :thinking:

---

Seems not:
```
❯ cargo run -- init bash | coreutils cat -A
M-oM-;M-?^M$
            __main() {^M$
                local major="${BASH_VERSINFO[0]}"^M$
                local minor="${BASH_VERSINFO[1]}"^M$
^M$
                if ((major > 4)) || { ((major == 4)) && ((minor >= 1)); }; then^M$
                    source <("C:\Users\david\Documents\GitHub\starshippppp\target\debug\starship.exe" init bash --print-full-init)^M$
                else^M$
                    source /dev/stdin <<<"$("C:\Users\david\Documents\GitHub\starshippppp\target\debug\starship.exe" init bash --print-full-init)"^M$
                fi^M$
            }^M$
            __main^M$
            unset -f __main^M$
            ^M$
```

---

Hm, I guess I was too optimistic 🤷